internal/pkg/service/buffer/storage/local/writer/volumes_test.go

```diff
@@ -0,0 +1,408 @@
+			FileOpenedAt:    randomFed2,
+			AllVolumes:      []string{"hdd/1", "hdd/2", "hdd/3", "ssd/1", "ssd/2", "top/1", "drained/1"},
+			ExpectedVolumes: []string{"hdd/2", "hdd/3", "ssd/1", "hdd/1", "top/1", "ssd/2"},
+		},
+		{
+			Name:            "count=10,pref=top,hdd,rand=1",
+			Count:           10,
+			PreferredTypes:  []string{"top", "hdd"},
+			FileOpenedAt:    randomFed1,
+			AllVolumes:      []string{"hdd/1", "hdd/2", "hdd/3", "ssd/1", "ssd/2", "top/1", "drained/1"},
+			ExpectedVolumes: []string{"top/1", "hdd/3", "hdd/1", "hdd/2", "ssd/2", "ssd/1"},
+		},
+		{
+			Name:            "count=10,pref=top,hdd,rand=2",
+			Count:           10,
+			PreferredTypes:  []string{"top", "hdd"},
+			FileOpenedAt:    randomFed2,
+			AllVolumes:      []string{"hdd/1", "hdd/2", "hdd/3", "ssd/1", "ssd/2", "top/1", "drained/1"},
+			ExpectedVolumes: []string{"top/1", "hdd/2", "hdd/3", "hdd/1", "ssd/1", "ssd/2"},
+		},
+		{
+			Name:            "count=10,pref=top,hdd,rand=3",
+			Count:           10,
+			PreferredTypes:  []string{"top", "hdd"},
+			FileOpenedAt:    randomFed3,
+			AllVolumes:      []string{"hdd/1", "hdd/2", "hdd/3", "ssd/1", "ssd/2", "top/1", "drained/1"},
+			ExpectedVolumes: []string{"top/1", "hdd/3", "hdd/2", "hdd/1", "ssd/2", "ssd/1"},
+		},
+	}
+
+	// Run test cases
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+
+			require.Greater(t, len(tc.AllVolumes), 0)
+
+			// Create volumes
+			volumesPath := t.TempDir()
+			createVolumes(t, volumesPath, tc.AllVolumes)
+
+			// Open volumes
+			ctx := context.Background()
+			logger := log.NewDebugLogger()
+			clk := clock.New()
+			volumes, err := OpenVolumes(ctx, logger, clk, volumesPath, WithWatchDrainFile(false))
+			require.NoError(t, err)
+
+			// Create a test file according to the test case specification
+			file := newStorageFile(t, tc.FileOpenedAt)
+			file.LocalStorage.Volumes.PerPod = tc.Count
+			file.LocalStorage.Volumes.PreferredTypes = tc.PreferredTypes
+
+			// Assign volume
+			fileVolumes := volumes.VolumesFor(file)
+
+			// Get IDs of the assigned volumes
+			actualVolumes := make([]string, len(fileVolumes))
+			for i, vol := range fileVolumes {
+				actualVolumes[i] = vol.ID().String()
+			}
+
+			// Compare
+			assert.Equal(t, tc.ExpectedVolumes, actualVolumes)
+
+			// Close volumes
+			assert.NoError(t, volumes.Close())
+		})
+	}
+}
+
+// createVolumes directories with ID file.
+func createVolumes(t *testing.T, volumesPath string, volumes []string) {
+	t.Helper()
+	for _, definition := range volumes {
+		require.Equal(t, 1, strings.Count(definition, "/"))
+		path := filepath.Join(volumesPath, filepath.FromSlash(definition))
+		assert.NoError(t, os.MkdirAll(path, 0o750))
+		assert.NoError(t, os.WriteFile(filepath.Join(path, volume.IDFile), []byte(definition), 0o640))
+
+		if strings.HasPrefix(definition, "drained/") {
+			require.NoError(t, os.WriteFile(filepath.Join(path, drainFile), []byte{}, 0o640))
+		}
+	}
+}
```

Comment:
v teste, `drained` volume type obsahuje vzdy iba `drained` volumes, je to podmienka na riadku `362`,